### PR TITLE
[Samba] Remove lo interface

### PR DESCRIPTION
**Pull request by lmagyar**
As suggested in #2447
Fixes v9.6.0, not like #2456

### CODE REVIEW

**Comment by lmagyar** [comment]
OK, I will emit an error in the log, but what should happen? Should the add-on start on all interfaces or fail to start?

File: samba/rootfs/etc/cont-init.d/samba.sh [hunk 1]
```diff
@@ -24,7 +24,6 @@ bashio::log.info "Hostname: ${HOSTNAME}"
 for interface in $(bashio::network.interfaces); do
     interfaces+=("${interface}")
 done
```
Comment (mdegat01) [inline samba/rootfs/etc/cont-init.d/samba.sh:26]:
```suggestion
interfaces+=($(bashio::network.interfaces))
if (( ${#interfaces[@]} == 0 )); then
    bashio::exit.nok "No supported interfaces found to bind on"
fi
```

**Review by mdegat01 — CHANGES_REQUESTED** [review]

File: samba/rootfs/etc/cont-init.d/samba.sh [hunk 2]
```diff
@@ -24,7 +24,15 @@ bashio::log.info "Hostname: ${HOSTNAME}"
...
-interfaces+=("lo")
+if [ ${#interfaces[@]} -eq 0 ]; then
+    bashio::log.fatal
+    bashio::log.fatal 'There are no supported interfaces to run Samba on.'
+    bashio::log.fatal
+    bashio::log.fatal 'Running Samba add-on on an unsupported installation'
+    bashio::log.fatal 'is not allowed.'
```
[the remark this answers is not in the document]
Comment (lmagyar) [reply samba/rootfs/etc/cont-init.d/samba.sh:32]: OK!

**Review by mdegat01 — APPROVED** [review]
Just tried it out on my dev system, all good. Thanks @lmagyar 👍

**Pull request merged**